### DISCUSSION TITLE
Using existing types in actions

### DISCUSSION THREAD

**Discussion by tombrouwer** [hasura/graphql-engine#8686] [Question] [answered]
Hi all, and thanks for the hard work.

I did setup a Hasura instance with Docker. I created some database tabels. One of them is called `list`. I want to create a action similar to insert_list, zo I can create a validation layer between the database and the action. When I want to create a action, it tells me some types not existing. Is it possible to use existing types which already exists form de database resources. Right now it throws an error that it is not existing. 

Action like:

```
type Mutation {
 insert_list(
    """the rows to be inserted"""
    objects: [list_insert_input!]!

    """upsert condition"""
    on_conflict: list_on_conflict
  ): list_mutation_response
}
```

Error:
```
\"list_insert_input\" is not defined in custom types or it is not a scalar/enum/input_object
```

**Comment by adas98012** [comment, marked as answer]
I have created a Github [issue](https://github.com/hasura/graphql-engine/issues/8881) . Please add the steps to reproduce and any other details/screenshots that will help us investigate the problem.

**Comment by soorajshankar** [comment]
Hey @adas98012, Thanks for starting the discussion! 
I added a comment on the issue that you have created [here](https://github.com/hasura/graphql-engine/issues/8881#issuecomment-1243543689)
